gym_xiangqi/envs/xiangqi_env.py

```diff
@@ -124,7 +137,59 @@ def __init__(self, agent_color=RED):
         self.game = XiangQiGame()
 
     def step(self, action):
-        return 0, 0, 0, 0
+        """
+        Run one time step of Xiangqi game: agent and enemy each plays a move
+        Parameter:
+            action (int): a valid action in Xiangqi action space
+        Return:
+            observation (object): agent's observation of the current environment
+            reward (float) : amount of reward returned after given action
+            done (bool): whether the episode has ended, in which case further
+                         step() calls will return undefined results
+            info (dict): contains auxiliary diagnostic information (helpful for
+                         debugging, and sometimes learning)
+        """
+        error_msg = "%r (%s) invalid action" % (action, type(action))
+        assert self.action_space.contains(action), error_msg
+
+        if self._done:
+            if not self.done_warn:
+                self.done_warn = True
+                print(gym.utils.colorize(
+                    "WARN: Environment should be reset with call to reset() "
+                    "when the episode has terminated (i.e 'done == True')",
+                    "yellow"
+                ))
+            return self.state, 0, self._done, {}
+
+        # initialize variables
+        reward = 0.0
+        pieces = self.agent_piece if self.turn == AGENT else self.enemy_piece
+
+        # if illegal move is given, penalize agent
+        if self.possible_actions[action] == 0:
+            return self.state, ILLEGAL_MOVE, False, {}
+
+        # if legal move, move the piece
+        piece, start, end = action_space_to_move(action)
+        pieces[piece].move(*end)
+
+        # update observation space
+        self.state[start[0]][start[1]] = EMPTY
+        removed_piece = self.state[end[0]][end[1]]
+        self.state[end[0]][end[1]] = piece * self.turn
+
+        # TODO: Implment conditions for checks and checkmates
+        #       and update done flag correspondingly
+
+        # reward based on removed piece
+        reward += PIECE_POINTS[abs(removed_piece)]
+
+        # self-play: agent switches turns between agent and enemy side
+        self.turn = ENEMY if self.turn == AGENT else AGENT
```

Comment:
Hmm yea I think we can focus on agent vs agent for now. Either way though we will need to swap the turn each time step method is called, since player moves will eventually call this same method also.